### ISSUE TITLE
VB SemanticModel fails to bind type argument inside cref even though compiler produces no error

### ISSUE THREAD

**Issue by shyamnamboodiripad** [dotnet/roslyn#2197]
The C# compiler (SemanticModel) correctly returns a ```TypeParameter``` symbol for the ```U``` in ```<see cref="C{U}"/>``` in the below example. I can validate this in the syntax visualizer (see below screenshot) and this also works via API (for example if I get symbol for ```C{U}``` and inspect its ```symbol.TypeArguments```.
![capture](https://cloud.githubusercontent.com/assets/10579684/7289053/5d9eafe2-e91f-11e4-924a-e89a6ae820ca.PNG)

For the corresponding case in VB, ```U``` does not bind (see below screenshot). If I get symbol for ```C(Of U)``` and inspect its ```symbol.TypeArguments```, I get back a ```ErrorTypeSymbol```. 
![capture2](https://cloud.githubusercontent.com/assets/10579684/7289089/f7816dfc-e91f-11e4-8607-070df98eb21a.PNG)

Shouldn't the VB compiler be able to return a valid symbol in the above case (considering that the code didn't produce any errors)? Why do the two compilers behave differently here.
